bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -1215,7 +1215,7 @@ private boolean isWritesetWritable(DistributionSchedule.WriteSet writeSet,
 
         int nonWritableCount = 0;
         for (int i = 0; i < sz; i++) {
-            if (!bk.getBookieClient().isWritable(getLedgerMetadata().currentEnsemble.get(i), key)) {
+            if (!bk.getBookieClient().isWritable(getCurrentEnsemble().get(i), key)) {
```

Comment:
Sample currentEnsemble before starting the loop?

Reply:
I am also in favor of saving the ensemble to a variable before starting the loop. However I think the logic here needs to be revisited, just to make sure ensemble changes won't impact the logic here.

Reply:
changed. it won't affect the logic. this code is an advisary thing to apply backpressure. The ensemble could still change between a writeset being seen as writable, and the pendingAddOp being triggered, but this won't cause a problem.